### PR TITLE
Improve look for student statistics page

### PR DESCRIPTION
**Pull request by kxmbrian**

### CODE REVIEW

File: app/views/course/statistics/_table.html.slim [hunk 1]
```diff
@@ -1,16 +1,28 @@
...
   - students.each_with_index do |student, index|
    = content_tag_for(:tr, student)
-     td = index + 1
-     td = link_to_user(student)
-     td = @service.tutors_of(student).map(&:name).join(', ')
-     td = student.level_number
-     td = link_to student.experience_points,
-                  course_user_experience_points_records_path(current_course, student)
+     td.text-center
+       = index + 1
```
Comment (weiqingtoh) [inline app/views/course/statistics/_table.html.slim]: Sorry I didn't catch this earlier - actually in line 15 you can do `- students.each.with_index(1) do |student, index|` and you need to do `= index + 1` here. One of those cool ruby enumerable methods.

File: app/services/course/student_statistics_service.rb [hunk 2]
```diff
@@ -18,11 +18,15 @@ def initialize(tutors)
...
+      group_managers_hash[group]
+    end.flatten.map(&:course_user).uniq
+  end
+
+  # Maps groups to their managers
+  #
+  # @return [Hash{Course::Group => Array<Course::GroupUser>}]
+  def group_managers_hash
+    @group_managers_hash ||=
+      @tutors.map(&:group_users).flatten.select(&:manager?).group_by(&:group)
```
Comment (weiqingtoh) [inline app/services/course/student_statistics_service.rb]: Not sure whether it matters if we use `group_by(&:group)` vs `group_by(&:group_id)`?

File: app/views/course/statistics/_table.html.slim (hunk 1 above)
Comment (allenwq) [inline app/views/course/statistics/_table.html.slim]: 👍 TIL

File: app/services/course/student_statistics_service.rb (hunk 2 above)
Comment (kxmbrian) [inline app/services/course/student_statistics_service.rb]: `:group` seems to work, but will change it to `:group_id` to be safe.

Comment (allenwq) [inline app/services/course/student_statistics_service.rb]: Think it's ok, group id should also have much better performance.

File: app/services/course/student_statistics_service.rb [hunk 3]
```diff
@@ -18,11 +18,15 @@ def initialize(tutors)
...
-      end
-    end.flatten.uniq
+      group_managers_hash[group.id]
+    end.flatten.map(&:course_user).uniq
+  end
+
+  # Maps groups to their managers
+  #
+  # @return [Hash{Course::Group => Array<Course::GroupUser>}]
+  def group_managers_hash
```
Comment (weiqingtoh) [inline app/services/course/student_statistics_service.rb]: Should we make this private? Or doesn't matter?

Comment (kxmbrian) [inline app/services/course/student_statistics_service.rb]: This method is called in the view to decide if the Tutors column should be displayed.

**Comment by kxmbrian** [comment]
Latest update has two changes:
- Added another method to the service and made `group_managers_hash` private.
- Removed the term 'tutor' form the service since the term can be ambiguous and the service might be used in a different context.

**Comment by allenwq** [comment]
LGTM, @weiqingtoh any more comments ?

**Pull request merged**